### PR TITLE
[Merged by Bors] - feat(SetTheory/Cardinal/Aleph): define initial ordinals

### PR DESCRIPTION
**Pull request by vihdzp**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/SetTheory/Ordinal/Basic.lean [hunk 1]
```diff
@@ -1469,3 +1503,5 @@ theorem List.Sorted.lt_ord_of_lt [LinearOrder α] [IsWellOrder α (· < ·)] {l
       | head as => exact List.head_le_of_lt hmltl
       | tail b hi => exact le_of_lt (lt_of_lt_of_le (List.rel_of_sorted_cons hm _ hi)
           (List.head_le_of_lt hmltl))
+
+set_option linter.style.longFile 1700
```
Comment (Command-Master) [inline Mathlib/SetTheory/Ordinal/Basic.lean:1507]: Is there no part of this file which could be split to a different one?

Comment (vihdzp) [reply Mathlib/SetTheory/Ordinal/Basic.lean:1507]: I can see three such things. 
- That weird lemma on lists has no reason to be here (and I deprecate it in #16966). 
- `Cardinal.univ` also shouldn't be defined here (and it should be defined as `#Cardinal` instead). 
- Finally, addition and successors should be moved to `Arithmetic` (which will also be a few hundred lines shorter once I deprecate stuff like `bsup`, `lsub`, etc.). 
All of these will take a bit of time, though.

Comment (vihdzp) [reply Mathlib/SetTheory/Ordinal/Basic.lean:1507]: Of course, the simpler alternative is to move the stuff about initial ordinals to `Cardinal/Ordinal`. But that file is very close to the 1.5k limit itself, so it'd be trading one thing for the other.

**Review by YnirPaz — APPROVED** [review]
Looks good!

File: Mathlib/SetTheory/Ordinal/Basic.lean (hunk 1 above)
Comment (YnirPaz) [reply Mathlib/SetTheory/Ordinal/Basic.lean:1507]: What about the 200 lines porting note at the end of `Cardinal/Ordinal`? Regardless we should add a file for more advanced cardinal stuff soon, as `Cardinal/Basic` is bloated as well.

Comment (vihdzp) [reply Mathlib/SetTheory/Ordinal/Basic.lean:1507]: Yeah, that porting note should be removed. Another idea I've had is splitting `Cardinal/Ordinal` into `Cardinal/Aleph` and `Cardinal/Arithmetic`.

Comment (vihdzp) [reply Mathlib/SetTheory/Ordinal/Basic.lean:1507]: Actually, I've made the first of these splits in #16990. I'll move this definition to the `Aleph` file if the refactor goes through.

**Comment by vihdzp** [comment]
I'll move these results once #16990 goes through.

File: Mathlib/SetTheory/Cardinal/Aleph.lean [hunk 2]
```diff
@@ -33,10 +33,70 @@ open Function Set Cardinal Equiv Order Ordinal
...
+/-- An ordinal is initial when it is the first ordinal with a given cardinality.
+
+This is written as `o.card.ord = o`, i.e. `o` is the smallest ordinal with cardinality `o.card`. -/
+def IsInitial (o : Ordinal) : Prop :=
+  o.card.ord = o
+
+theorem IsInitial.ord_card {o : Ordinal} (h : IsInitial o) : o.card.ord = o := h
+
+theorem IsInitial.le_of_card_le {a b : Ordinal} (ha : IsInitial a)
+    (hb : a.card ≤ b.card) : a ≤ b := by
```
Comment (fpvandoorn) [inline Mathlib/SetTheory/Cardinal/Aleph.lean:49]: make `iff`

File: Mathlib/SetTheory/Cardinal/Aleph.lean [hunk 3]
```diff
@@ -33,10 +33,70 @@ open Function Set Cardinal Equiv Order Ordinal
...
+  rw [← ha.ord_card]
+  exact (ord_mono hb).trans (ord_card_le b)
+
+theorem isInitial_ord (c : Cardinal) : IsInitial c.ord := by
+  rw [IsInitial, card_ord]
+
+theorem IsInitial.card_le_card {a b : Ordinal} (ha : IsInitial a) (hb : IsInitial b) :
```
Comment (fpvandoorn) [inline Mathlib/SetTheory/Cardinal/Aleph.lean:56]: This should hold without `hb`.

**Review by fpvandoorn — COMMENTED** [review]
bors d+

**Pull request closed without merging**